### ISSUE TITLE
CronJob Name Too Long - must be no more than 52 characters.

### ISSUE THREAD

**Issue by medined** [helm/monocular#566]
I ran a helm command to install monoclar:

```
./helm install --namespace office  monocular/monocular
```

which resulted in the following message.

```
Error: release wrapping-grasshopper failed: CronJob.batch "wrapping-grasshopper-monocular-sync-scheduled-incubator" is invalid: metadata.name: Invalid value: "wrapping-grasshopper-monocular-sync-scheduled-incubator": must be no more than 52 characters
```

**Comment by mattiaperi** [comment]
Hi all, as @medined, I'm facing the same issue:
`$ helm install monocular/monocular
Error: release terrifying-grasshopper failed: CronJob.batch "terrifying-grasshopper-monocular-sync-scheduled-incubator" is invalid: metadata.name: Invalid value: "terrifying-grasshopper-monocular-sync-scheduled-incubator": must be no more than 52 characters`

`$ echo "terrifying-grasshopper-monocular-sync-scheduled-incubator" | wc -m
      58`

Obviously a:

`$ helm install --name monocular monocular/monocular`

forcing the release name less than 14 characters it's a workaround for the issue.

**Comment by Dean-Coakley** [comment]
@mattiaperi I am working on resolving this issue, however I'm getting.
`must be no more than 63 characters`
and have validated that it works for me up to 63 characters. I installed a release with name of 18 chars.

Are both of you possibly on an old helm release?
```
$ helm version
Client: &version.Version{SemVer:"v2.13.0", GitCommit:"79d07943b03aea2b76c12644b4b54733bc5958d6", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.0", GitCommit:"79d07943b03aea2b76c12644b4b54733bc5958d6", GitTreeState:"clean"}
```

**Comment by Dean-Coakley** [comment]
Nevermind, if the name is far too long it will fail on `spec.template.labels`
Otherwise it will fail on 
`metadata.name` of the cronjob at 52 chars as you've described

**Comment by harindersingh** [comment]
If you're using helm 3,
`--name`
flag is invalid and will give
`Error: unknown flag: --name`

You directly provide name before chartname
`Usage:
  helm install [NAME] [CHART] [flags]`
`helm install monocular-release monocular/monocular`

**Comment by Dean-Coakley** [comment]
Right, I'll try update the chart docs later today if I get a chance.

However that's unrelated to this issue:
`./helm install this-is-very-long-release-name chart/monocular`
Will still result in a Crontab with a name too long resulting in an invalid install.
